subcommand.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+)
+
+type SubCommand struct {
+	cmd      *exec.Cmd
+	PipeRead *os.File
+}
+
+type Stream int
+
+const (
+	StreamStdout Stream = 1 << iota
+	StreamStderr
+)
+
+func NewSubCommand(args []string, stream Stream) SubCommand {
+	pipeRead, pipeWrite, err := os.Pipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	env := append(os.Environ(), "GODEBUG=gctrace=1")
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = env
+	cmd.Stdin = os.Stdin
+
+	switch stream {
+	case StreamStdout:
+		cmd.Stdout = pipeWrite
```

Comment:
Something is wrong with this logic, if you call a command which fails, and writes to stderr, stderr is lost. At master stderr is forwarded through to the caller.